integreat_cms/cms/views/utils/search_content_ajax.py

```diff
@@ -105,6 +106,34 @@ def search_content_ajax(  # noqa: PLR0915, PLR0912, C901
     results: list[dict[str, Any]] = []
 
     user = request.user
+
+    if "contact" in object_types:
```

Comment:
I know that you're already working on a refactoring for this, but maybe this is still helpful. 
I think this function (only the contact section) by itself already does a lot. For example I think it would be good to separate logical units and would extract the check for permissions to another function. But as said, I'm going to let it pass during this review :)